geomesa-convert/geomesa-convert-common/src/main/scala/org/locationtech/geomesa/convert/SimpleFeatureValidator.scala

```diff
@@ -0,0 +1,173 @@
+
+  override def lastError: String = lastErr
+}
+
+class HasGeoValidator extends SimpleFeatureValidator {
+  private var valid: (SimpleFeature) => Boolean = _
+  private var lastErr: String = _
+
+  override def name: String = "has-dtg"
+
+  override def init(sft: SimpleFeatureType): Unit = {
+    valid = {
+      lastErr = null
+      if (sft.getGeometryDescriptor != null) {
+        (sf: SimpleFeature) =>
+          val res = sf.getDefaultGeometry != null
+          if (!res) {
+            lastErr = "Null date attribute"
+          }
+          res
+      } else {
+        (_) => true
+      }
+    }
+  }
+
+  override def validate(sf: SimpleFeature): Boolean = valid(sf)
+
+  override def lastError: String = lastErr
+}
+
+/**
+  * Validates that the geometry fits into a Z2 or Z3 style WGS84 index scheme
+  */
+class ZIndexValidator extends SimpleFeatureValidator {
+  private var valid: (SimpleFeature) => Boolean = _
+  private var lastErr: String = _
+
+  override def name: String = "z-index"
+
+  override def init(sft: SimpleFeatureType): Unit = {
+    valid = {
+      import org.locationtech.geomesa.utils.geotools.Conversions._
+      import org.locationtech.geomesa.utils.geotools.RichSimpleFeatureType.RichSimpleFeatureType
+
+      val maxDate = BinnedTime.maxDate(sft.getZ3Interval).toDate
+      val dtgFn: (SimpleFeature) => Boolean = sft.getDtgIndex match {
+        case Some(dtgIdx) => (sf: SimpleFeature) =>
+          lastErr = null
+          if (sf.getAttribute(dtgIdx) == null){
+            lastErr = "Null date attribute"
+            false
+          } else if (sf.get[Date](dtgIdx).before(Z3IndexValidator.Z3MinDate)) {
+            lastErr = "Date is before Z3 Min Date"
+            false
+          } else if (sf.get[Date](dtgIdx).after(maxDate)) {
+            lastErr = "Date is after Z3 Max Date"
+            false
+          } else {
+            true
+          }
+        case None =>
+          throw new IllegalArgumentException("Z3 validator cannot be used on a type lacking a dtg index")
+      }
+
+      val geomFn: (SimpleFeature) => Boolean =
+        if (sft.getGeometryDescriptor != null) {
+          (sf: SimpleFeature) =>
+            lastErr = null
+            if (sf.getDefaultGeometry == null) {
+              lastErr = "Geom is null"
+              false
+            } else if (!org.locationtech.geomesa.utils.geotools.wholeWorldEnvelope.contains(sf.geometry.getEnvelopeInternal)) {
+              lastErr = "Geom is not contained in whole world envelope"
+              false
+            } else {
+              true
+            }
+        } else {
+          throw new IllegalArgumentException("Z3 validator cannot be used on a type lacking a dtg index")
+        }
+
+      (sf: SimpleFeature) => dtgFn(sf) && geomFn(sf)
+    }
+  }
+
+  override def validate(sf: SimpleFeature): Boolean = valid(sf)
+
+  override def lastError: String = lastErr
+}
+
+object Z3IndexValidator {
+  val Z3MinDate = new Date(0)
```

Comment:
it might be good to move this into the BinnedTime class - there was some discussion about supporting dates before the epoch, and we'd probably forget to come back and update this if we ever implemented that.